### DISCUSSION TITLE
docker compose 2.32 breaking doodba

### DISCUSSION THREAD

**Discussion by ap-wtioit** [Tecnativa/doodba#640] [Show and tell]
Heads up before christmas as this might not be fixed in time

See: https://github.com/docker/compose/issues/12402

Currently the way we use `extends:` and `depends_on:` is broken in docker compose 2.32. Preventing the startup of services that odoo depends on.

To downgrade docker compose on debian(ish):
* `apt-cache madison docker-compose-plugin`
   should give something like
   ```
   docker-compose-plugin | 2.32.1-1~ubuntu.24.04~noble | https://download.docker.com/linux/ubuntu noble/stable amd64 Packages
   docker-compose-plugin | 2.31.0-1~ubuntu.24.04~noble | https://download.docker.com/linux/ubuntu noble/stable amd64 Packages
   ...
   ```
* `sudo apt install docker-compose-plugin=2.31.0-1~ubuntu.24.04~noble` pick a non 2.23  version

**Comment by ap-wtioit** [comment]
downgraded our servers with ansible:
```bash
ansible -m shell -a "apt install -y --allow-downgrades docker-compose-plugin=\$(apt-cache madison docker-compose-plugin | grep -v '2.32.' | head -n 1 | cut -d' ' -f3)" build_servers --become
```

**Comment by josep-tecnativa** [comment]
Hi Andreas, thank you for the heads-up. We are currently using Docker Compose v2.29, so it’s not causing any issues for us at the moment. However, we appreciate the notice and will keep an eye on the bug to avoid updating until it’s resolved.

**Reply by ap-wtioit** [reply]
yeah, just thought it might be helpful to others to know about this. As my workstation (Ubuntu 24.04) updated today in the morning and the errors were not clear why they happened.

**Comment by josep-tecnativa** [comment]
It seems they fixed the issue a few minutes ago. https://github.com/compose-spec/compose-go/pull/721

**Reply by ap-wtioit** [reply]
yes, now we need to wait for the release to happen and propagate through the repos (might take some time)
